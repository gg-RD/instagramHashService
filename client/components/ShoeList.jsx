import React from 'react';

import ShoeItem from './ShoeItem.jsx';


export default class ShoeList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      show_remaining: false,
    };
  }

  render(){
    return(
      <div className = 'shoes'>
        <table className='table'>
          <tr id='table-row'>
          {this.props.list1.map((i) =>
            <ShoeItem key={i.id} item={i}/>
          )}
          </tr>
          <tr id='table-row' className={!this.state.show_remaining ?'hidden':''}>
          {this.props.list2.map((i) =>
            <ShoeItem key={i.id} item={i}/>
          )}
          </tr>
          <tr id='table-row' className={!this.state.show_remaining ?'hidden':''}>
          {this.props.list3.map((i) =>
            <ShoeItem key={i.id} item={i}/>
          )}
          </tr>
        </table>
        <div className='show-remaining' onClick={() => {
          var newState = !this.state.show_remaining;
          this.setState({show_remaining:newState});
        }}>
          {!this.state.show_remaining ?
            <div>
              <a className='show-remaining_underline'>Load More ({this.props.list2.length +
              this.props.list3.length})</a>
              <a>∨</a>
            </div>
            :
            <div>
              <a className='show-remaining_underline'>Hide</a> <a>∧</a>
            </div>
          }
        </div>
    </div>);
  };

}
