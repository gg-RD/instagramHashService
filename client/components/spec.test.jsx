import React from 'react';
import Info from './iteminfo.jsx';
import ShoeList from './iteminfo.jsx';
import { mount } from 'enzyme';
import {shallow} from 'enzyme';
import sinon from 'sinon';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
configure({ adapter: new Adapter() });

var exampleData = [ { id: '1827918187418364500_8255024479',
    user:
     { id: '8255024479',
       full_name: 'hack reactor gg',
       profile_picture: 'https://instagram.fdac10-1.fna.fbcdn.net/vp/eaf4663b993b22ab3c90681222cba10e/5C0B007A/t51.2885-19/11906329_960233084022564_1448528159_a.jpg',
       username: 'hackreactorggrd' },
    images:
     { thumbnail: [Object],
       low_resolution: [Object],
       standard_resolution: [Object] },
    created_time: '1532124850',
    caption:
     { id: '17941740733080631',
       text: '#AirMax98',
       created_time: '1532124850',
       from: [Object] },
    user_has_liked: true,
    likes: { count: 1 },
    tags: [ 'airmax98' ],
    filter: 'Normal',
    comments: { count: 0 },
    type: 'image',
    link: 'https://www.instagram.com/p/BleEegfjCpU/',
    location: null,
    attribution: null,
    users_in_photo: [] },
  { id: '1827798596344187322_8255024479',
    user:
     { id: '8255024479',
       full_name: 'hack reactor gg',
       profile_picture: 'https://instagram.fdac10-1.fna.fbcdn.net/vp/eaf4663b993b22ab3c90681222cba10e/5C0B007A/t51.2885-19/11906329_960233084022564_1448528159_a.jpg',
       username: 'hackreactorggrd' },
    images:
     { thumbnail: [Object],
       low_resolution: [Object],
       standard_resolution: [Object] },
    created_time: '1532110593',
    caption:
     { id: '17947938481112491',
       text: '#AirMax97',
       created_time: '1532110593',
       from: [Object] },
    user_has_liked: true,
    likes: { count: 5 },
    tags: [ 'airmax97' ],
    filter: 'Normal',
    comments: { count: 0 },
    type: 'image',
    link: 'https://www.instagram.com/p/BldpSOoDYm6/',
    location:
     { latitude: 37.78371,
       longitude: -122.40896,
       name: 'Hack Reactor',
       id: 70325290 },
    attribution: null,
    users_in_photo: [] } ]


describe('<Info />', () => {
  it('calls componentDidMount', () => {
      sinon.spy(Info.prototype, 'componentDidMount');
      var wrapper = mount(<Info />);
      expect(Info.prototype.componentDidMount.calledOnce).toBe(true);
      wrapper.unmount();
    });

});

describe('<ShoeList />', () => {
  it('allows us to set props and click event', () => {
    var onButtonClick = sinon.spy()
    var wrapper = mount(<ShoeList
      list1={exampleData}
      list2={exampleData}
      list3={exampleData}
      list={exampleData}
      onButtonClick={onButtonClick}
      />);
    expect(wrapper.find('.flex-container').childAt(2).props().id).toBe('hidden');
    console.log(wrapper.find('.flex-container').childAt(2).props().id);
    wrapper.find('.show-remaining').simulate('click');
    console.log(wrapper.find('.flex-container').childAt(2).props().id);
    expect(wrapper.find('.flex-container').childAt(2).props().id).toMatch('flex-container3');
  });


});
