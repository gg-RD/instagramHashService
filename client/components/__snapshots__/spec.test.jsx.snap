// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  style={
    Object {
      "borderWidth": "2px",
      "float": "left",
      "height": "35%",
      "textAlign": "center",
      "width": "35%",
    }
  }
>
  <h3
    className="title"
  >
     HOW OTHERS ARE WEARING IT 
  </h3>
  <h5
    className="subtitle"
  >
    Mention @Nike on Instagram for a chance to have your look featured.
  </h5>
  <div
    className="flex-container"
  >
    <div
      id="flex-container1"
    />
    <div
      id="hidden"
    />
    <div
      id="hidden"
    />
    <div
      className="show-remaining"
      onClick={[Function]}
    >
      <div
        className="show-remaining_underline"
      >
        <a>
          Load More (
          0
          )
        </a>
        <a>
          ∨
        </a>
      </div>
    </div>
  </div>
  <div
    id="overlay"
  >
    <div
      className="carousel"
    >
      <div
        id="left_arrow"
      >
        <div
          className="slide-arrow left"
          onClick={[Function]}
        >
          ◀
        </div>
      </div>
      <div>
        <div
          className="white-box"
        >
          <img
            className="image_in_box"
            src={undefined}
          />
          <div>
            <button
              aria-label="Close"
              className="close_button"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
            <img
              id="user_profile_pic"
              src={undefined}
            />
            <div
              id="user_box"
            >
              <p
                id="insta_name"
              >
                 
                 
              </p>
              <p
                id="creation_time"
              >
                NaN
                 
                days
                 ago 
              </p>
            </div>
          </div>
          <div
            className="gray_line"
          >
            <p
              id="In_look"
            >
               In this Look 
            </p>
          </div>
        </div>
      </div>
      <div
        id="r_arrow"
      >
        <div
          className="slide-arrow right"
          onClick={[Function]}
        >
          ▶
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Link changes the class when hovered 2`] = `
<div
  style={
    Object {
      "borderWidth": "2px",
      "float": "left",
      "height": "35%",
      "textAlign": "center",
      "width": "35%",
    }
  }
>
  <h3
    className="title"
  >
     HOW OTHERS ARE WEARING IT 
  </h3>
  <h5
    className="subtitle"
  >
    Mention @Nike on Instagram for a chance to have your look featured.
  </h5>
  <div
    className="flex-container"
  >
    <div
      id="flex-container1"
    />
    <div
      id="hidden"
    />
    <div
      id="hidden"
    />
    <div
      className="show-remaining"
      onClick={[Function]}
    >
      <div
        className="show-remaining_underline"
      >
        <a>
          Load More (
          0
          )
        </a>
        <a>
          ∨
        </a>
      </div>
    </div>
  </div>
  <div
    id="overlay"
  >
    <div
      className="carousel"
    >
      <div
        id="left_arrow"
      >
        <div
          className="slide-arrow left"
          onClick={[Function]}
        >
          ◀
        </div>
      </div>
      <div>
        <div
          className="white-box"
        >
          <img
            className="image_in_box"
            src={undefined}
          />
          <div>
            <button
              aria-label="Close"
              className="close_button"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
            <img
              id="user_profile_pic"
              src={undefined}
            />
            <div
              id="user_box"
            >
              <p
                id="insta_name"
              >
                 
                 
              </p>
              <p
                id="creation_time"
              >
                NaN
                 
                days
                 ago 
              </p>
            </div>
          </div>
          <div
            className="gray_line"
          >
            <p
              id="In_look"
            >
               In this Look 
            </p>
          </div>
        </div>
      </div>
      <div
        id="r_arrow"
      >
        <div
          className="slide-arrow right"
          onClick={[Function]}
        >
          ▶
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Shows other list items when 1`] = `
<div
  style={
    Object {
      "borderWidth": "2px",
      "float": "left",
      "height": "35%",
      "textAlign": "center",
      "width": "35%",
    }
  }
>
  <h3
    className="title"
  >
     HOW OTHERS ARE WEARING IT 
  </h3>
  <h5
    className="subtitle"
  >
    Mention @Nike on Instagram for a chance to have your look featured.
  </h5>
  <div
    className="flex-container"
  >
    <div
      id="flex-container1"
    />
    <div
      id="hidden"
    />
    <div
      id="hidden"
    />
    <div
      className="show-remaining"
      onClick={[Function]}
    >
      <div
        className="show-remaining_underline"
      >
        <a>
          Load More (
          0
          )
        </a>
        <a>
          ∨
        </a>
      </div>
    </div>
  </div>
  <div
    id="overlay"
  >
    <div
      className="carousel"
    >
      <div
        id="left_arrow"
      >
        <div
          className="slide-arrow left"
          onClick={[Function]}
        >
          ◀
        </div>
      </div>
      <div>
        <div
          className="white-box"
        >
          <img
            className="image_in_box"
            src={undefined}
          />
          <div>
            <button
              aria-label="Close"
              className="close_button"
              onClick={[Function]}
              type="button"
            >
              <span
                aria-hidden="true"
              >
                ×
              </span>
            </button>
            <img
              id="user_profile_pic"
              src={undefined}
            />
            <div
              id="user_box"
            >
              <p
                id="insta_name"
              >
                 
                 
              </p>
              <p
                id="creation_time"
              >
                NaN
                 
                days
                 ago 
              </p>
            </div>
          </div>
          <div
            className="gray_line"
          >
            <p
              id="In_look"
            >
               In this Look 
            </p>
          </div>
        </div>
      </div>
      <div
        id="r_arrow"
      >
        <div
          className="slide-arrow right"
          onClick={[Function]}
        >
          ▶
        </div>
      </div>
    </div>
  </div>
</div>
`;
